integration_tests/features/steps/fal_steps.py

```diff
@@ -114,30 +115,38 @@ def invoke_failing_fal_flow(context):
 
 @then("the following scripts are ran")
 def check_script_files_exist(context):
-    scripts_exist = _check_files_exist(context, context.table.headings)
-    if not all(scripts_exist.values()):
-        not_existent = map(
-            lambda t: t[0], filter(lambda t: not t[1], scripts_exist.items())
-        )
-        to_report = ", ".join(not_existent)
-        assert False, f"Script files {to_report} should BE present"
+    python_scripts = _get_all_python_scripts(context.temp_dir.name)
+    expected_scripts = list(map(_script_filename, context.table.headings))
+    unittest.TestCase().assertCountEqual(python_scripts, expected_scripts)
 
 
 @then("the following scripts are not ran")
```

Comment:
So we were already making it exhaustive by hand when we needed to.

Reply:
> Removed some redundant cases of following scripts are not ran cases (if there is a preceding following scripts are ran check, that means this case is now redundant. Didn't remove all of them since there are still some valid usages).

Yep, I've removed the cases where we manually checked for this before.